src/NCalc.Core/Helpers/MathHelper.cs

```diff
@@ -485,4 +485,51 @@ private static void CheckOverflow(dynamic value)
                 throw new OverflowException("Arithmetic operation resulted in an overflow");
         }
     }
+
+    public static object Factorial(object? result)
+    {
+        switch (result)
+        {
+            case int v:
+                return CalculateFactorial(v);
+            case long v:
+                return CalculateFactorial(v);
+            case float v:
+                return CalculateFactorial(v);
+            case double v:
+                return CalculateFactorial(v);
+            case decimal v:
+                return CalculateFactorial(v);
+            case BigInteger v:
+                return CalculateFactorial(v);
+            default:
+                throw new ArgumentException("Unsupported numeric type.", nameof(result));
+        }
+    }
+
+#if NET
+    private static T CalculateFactorial<T>(T n) where T : INumber<T>
+    {
+        if (n < T.Zero)
+            throw new ArgumentOutOfRangeException(nameof(n));
+
+        var one = T.One;
+        var r = one;
+
+        for (var i = one + one; i <= n; i++)
+            r *= i;
+
+        return r;
+    }
+#else
+    private static dynamic CalculateFactorial(dynamic n)
+    {
```

Comment:
Can you please add negative value check to be more consistent with INumber variant?

Reply:
Sure